geyser-plugin-manager/src/geyser_plugin_manager.rs

```diff
@@ -20,26 +21,6 @@ impl GeyserPluginManager {
         }
     }
 
-    /// # Safety
-    ///
-    /// This function loads the dynamically linked library specified in the path. The library
-    /// must do necessary initializations.
-    pub unsafe fn load_plugin(
-        &mut self,
-        libpath: &str,
-        config_file: &str,
-    ) -> Result<(), Box<dyn Error>> {
-        type PluginConstructor = unsafe fn() -> *mut dyn GeyserPlugin;
-        let lib = Library::new(libpath)?;
-        let constructor: Symbol<PluginConstructor> = lib.get(b"_create_plugin")?;
-        let plugin_raw = constructor();
-        let mut plugin = Box::from_raw(plugin_raw);
```

Comment:
cc/ @OliverNChalk
bug introduced by the code that replaced this.

In Rust variables on stack will be dropped in reverse declaration order.
In this (removed) version of the function, library lives longer than the plugin.
In the new version, library is dropped first which will lead to sigsev.

Reply:
rip